apps/web/lib/checkForEmptyAssignment.ts

```diff
@@ -0,0 +1,30 @@
+import type { EventTypeAssignedUsers, EventTypeHosts } from "~/event-types/views/event-types-single-view";
+
+// This function checks if EventType requires assignment.
+// returns true: if EventType requires assignment but there is no assignment yet done by the user.
+// returns false: for all other scenarios.
+export function checkForEmptyAssignment({
+  assignedUsers,
+  hosts,
+  isManagedEventType,
+  assignAllTeamMembers,
+  isTeamEvent,
+}: {
+  assignedUsers: EventTypeAssignedUsers;
+  hosts: EventTypeHosts;
+  isManagedEventType: boolean;
+  assignAllTeamMembers: boolean;
+  isTeamEvent: boolean;
+}): boolean {
+  if (isTeamEvent && assignAllTeamMembers) {
+    return false;
+  } else if (
+    isTeamEvent &&
+    // For managed eventtype check if assigned users are empty.
+    // For non-managed eventtype check if hosts are empty.
+    (isManagedEventType ? assignedUsers.length === 0 : hosts.length === 0)
```

Comment:
Hey as @sean-brydon suggested that assignedUsers can be empty for ManagedEventType. So pls fix it

Reply:
No, he is trying to say to consider the scenario of ManagedEventType having empty assigned users.
Also , If we refer to current main branch code :
https://github.com/calcom/cal.com/blob/33e1abf4a7324b3bd2c9a0d5708379e51636234b/apps/web/modules/event-types/views/event-types-single-view.tsx#L412

Here warning dialog is shown if ManagedEventType and assignedusers are empty.

Reply:
Ahh, got it